### ISSUE TITLE
Consider input of unserialized values by default

### ISSUE THREAD

**Issue by medikoo** [medikoo/dbjs-persistence#4]
Default methods (`store`, `storeMany`) may take unserialized values,
and for storing using serialized values we may introduce (`storeRaw` `storeManyRaw`).

Similarily with `get` methods, by default they may return unserialized, and there cane be `*Raw` version to return serialized.

**Comment by medikoo** [comment]
It's probably not great idea to have methods which can work only with certain types of valid dbjs values, and will crash with other types of (still valid) dbjs values.

The other way would be to introduce easily accessible _serialize_, _unserialize_, _resolveId_ functions.
They maybe 1-3 letter named functions (`s`, `us`, `ri` ?) directly on _driver_ instance. That way user of driver will not have to deduct serialized/unserialized values manually, but could just fallback to provided utils.

It's nothing uncommon to use some escape/unescape (serialize/unserialize) methods when working with db drivers.

**Comment by medikoo** [comment]
The other idea that came out recently, we may have four different means of data resolution:

##### 1. Raw (serialized)

_Currently provided in this project_

Data is resolved from persistent layer in serialized low-level form as e.g.

```javascript
[
  { id: 'foo', data: { value: '7User#', stamp: 223423432 } },
  { id: 'foo/bar', data: { value: '11', stamp: 223423433 } },
  { id: 'foo/22', data: { value: '230', stamp: 223423433 } },
  { id: 'foo/address/street', data: { value: '3whatever', stamp: 223423433 } },
  { id: 'foo/objectProp', data: { value: '7otherObj', stamp: 123131312312 } }
]
```

##### 2.Unserialized (but with no nesteds resolution)

_Not provided at this point_

Out of a _storage_ we may get an adapter, that for same methods will resolve values in unserialized form and with no stamp information (as it's not needed for many cases). The solution for objects would be, that it would unserialize those values to plain hashes: `{ id: 'some/object/id' }`. Therefore it'll be easily to distinguish it from plain string values. Result of querying for an object data would be:

```javascript
{
  id: 'foo', // non-enumerable
  bar: true,
  '22': 30,
  'address/street': 'whatever',
  objectProp: { id: 'otherObj' }
}
```

##### 3. Unserialized (with nesteds resolution)

_Not provided at this point_

```javascript
{
  id: 'foo', // non-enumerable
  bar: true,
  '22': 30,
  address: { street 'whatever' },
  objectProp: { id: 'otherObj' }
}
```

##### 4. Fully resolved to Dbjs instance

_Provided in this project, via `load*` methods if properly configured dbjs instance was provided to driver_

Population of data to dbjs is the only solution that can provide access to fully resolved and validated values with stamps information.

@kamsi @mtuchowski what do you think?

**Comment by mtuchowski** [comment]
I personally like version 3 the most. It's basically a JSON!

**Comment by kamsi** [comment]
I agree that we should have version 3... probably we also should (AFTER we have 3) provide dbjs version as dbjs is ultimately our data mapper.
